commit/merkleroot/rmn/controller.go

```diff
@@ -1072,11 +1073,24 @@ func randomShuffle[T any](s []T) []T {
 	return ret
 }
 
-func newRequestID() uint64 {
+var (
+	// atomicCounter is used to generate unique request IDs.
+	atomicCounter = &atomic.Int64{}
+)
+
+// newRequestID generates a new unique request ID.
+func newRequestID(lggr logger.Logger) uint64 {
 	b := make([]byte, 8)
 	_, err := crand.Read(b)
 	if err != nil {
-		panic(err)
+		// fallback to time-based id in the very rare scenario that the random number generator fails
+		lggr.Warnw("failed to generate random request id, falling back to time based",
+			"err", err,
+			"atomicCounter", atomicCounter.Load(),
+		)
+		t := time.Now().UTC().UnixNano()
+		c := atomicCounter.Add(1)
```

Comment:
The end of the timestamp has the most entropy, it would be better to include the counter at the front.

Not sure this is strictly correct, but I'm thinking something like this
```
b := make([]byte, 8)
binary.BigEndian.PutInt64(b, t)
b[0]=byte(c%256)
```